apps/backend/src/datasources/postgres/TemplateDataSource.ts

```diff
@@ -720,27 +720,58 @@ export default class PostgresTemplateDataSource implements TemplateDataSource {
     question: string,
     default_config: string
   ): Promise<Question> {
-    const resultSet: QuestionRecord[] = await database
-      .insert(
-        {
-          category_id,
-          question_id,
-          natural_key,
-          data_type,
-          question,
-          default_config,
-        },
-        ['*']
-      )
-      .from('questions')
-      .onConflict('question_id')
-      .merge();
+    try {
+      const resultSet: QuestionRecord[] = await database
+        .insert(
+          {
+            category_id,
+            question_id,
+            natural_key,
+            data_type,
+            question,
+            default_config,
+          },
+          ['*']
+        )
+        .from('questions')
+        .onConflict('question_id')
+        .merge();
 
-    if (!resultSet || resultSet.length != 1) {
-      throw new GraphQLError('Failure to upsert question');
-    }
+      if (!resultSet || resultSet.length != 1) {
+        throw new GraphQLError('Failure to upsert question');
+      }
 
-    return createQuestionObject(resultSet[0]);
+      return createQuestionObject(resultSet[0]);
+    } catch (error) {
+      if (
+        (error as { constraint: string }).constraint ===
+        'questions_natural_key_key'
+      ) {
+        const resultSet: QuestionRecord[] = await database
+          .insert(
+            {
+              category_id,
+              question_id,
+              natural_key: natural_key.concat('_copy'),
+              data_type,
+              question,
+              default_config,
+            },
+            ['*']
+          )
+          .from('questions')
+          .onConflict('question_id')
+          .merge();
+
+        if (!resultSet || resultSet.length != 1) {
+          throw new GraphQLError('Failure to upsert question');
+        }
+
+        return createQuestionObject(resultSet[0]);
+      } else {
+        throw new GraphQLError('Failure to upsert question');
+      }
+    }

```

Comment:
If possible can we check first if key exists then change it rather than rely on a failure.

Reply:
Sure. Just to clarify, you are mentioning about the actual natural key which is coming as part of the template which we are trying to import? In that case its already being checked as part of the if condition right.
if (
        (error as { constraint: string }).constraint ===
        'questions_natural_key_key'
      )

Reply:
I am saying maybe we can check if any keys are already in the database then change them rather than try to insert then make changes when the operation fails.